### PR TITLE
Merge develop 8.1 30.05.17

### PR DESCRIPTION
**Pull request by livecodeali**
Conflicts:
- both modified:   engine/src/funcs.cpp
	- conflict due to MCContainer class changes. Kept new 8.1 code structure, but reinstated MCContainer instead of MCAutoPointer<MCContainer>
- both modified:   revdb/src/postgresql_connection.cpp
	- conflict due to removal of uninitialised variable in both branches. kept develop-8.1 changes.
- both modified:   tests/_testrunner.livecodescript
	- moved uncaught error dialog handling to correct place in _testrunnerbehavior
- both modified:   tests/lcs/core/engine/_extension.lcb
- both modified:   tests/lcs/core/engine/extension.livecodescript
	- additions in both branches - kept both.

### CODE REVIEW

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok 19b55b5

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok 19b55b51fbd1dd9786e250f520cb44db593b0064

**Comment by livecode-vulcan** [comment]
😞 test failure 19b55b51fbd1dd9786e250f520cb44db593b0064

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/314)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/320)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/300)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/243)
* try-community-universal-ios-iphonesimulator10.3: [pending](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/243)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/309)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/314)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/312)
* try-community-x86-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/318)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/312)
* try-community-x86_64-linux-debian8: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/318)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/337)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/308)

File: tests/lcs/core/engine/_extension.lcb [hunk 1]
```diff
@@ -50,6 +50,21 @@ public handler TestExtensionExecuteScript_Do(in pScript as String, in pTarget as
 		execute script pScript in the result with pParams
 	end if
 	return the result
+
```
Comment (livecodepanos) [inline tests/lcs/core/engine/_extension.lcb]: oh missed that, a `end handler` is missing :)

**Comment by livecodeali** [comment]
@livecodepanos Fixed, thanks :-)

**Comment by livecodepanos** [comment]
@livecode-vulcan review ok fe94afb

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok fe94afb50c15f69449d93e70ac343921f0a13d51

**Comment by livecode-vulcan** [comment]
😎 test success fe94afb50c15f69449d93e70ac343921f0a13d51

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/315)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/321)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/301)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/244)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/244)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/310)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/315)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/313)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/319)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/313)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/319)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/338)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/309)

**Comment by livecodeali** [comment]
@livecode-vulcan review by @livecodepanos ok d345ee8

Just amended to change the commit hash and force-pushed to retry Travis.

**Comment by livecode-vulcan** [comment]
💙 review by @livecodepanos ok d345ee8ca48cb03128cb58e96e4f017ddb8ea1e8

**Comment by livecode-vulcan** [comment]
😎 test success d345ee8ca48cb03128cb58e96e4f017ddb8ea1e8

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/318)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/324)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/304)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/247)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/247)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/313)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/318)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/316)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/322)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/316)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/322)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/341)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/312)

**Pull request merged**